packages/mui-codemod/src/deprecations/button-group-classes/button-group-classes.js

```diff
@@ -0,0 +1,127 @@
+                  );
+                  const precedingTemplateElement = parent.quasis[memberExpressionIndex];
+                  const atomicClasses = replacementSelector
+                    .replaceAll('MuiButtonGroup-', '')
+                    .replaceAll(replacementSelectorPrefix, '')
+                    .replaceAll(' > ', '')
+                    .split('.')
+                    .filter(Boolean);
+
+                  if (
+                    precedingTemplateElement.value.raw.endsWith(
+                      deprecatedClass.startsWith(' ')
+                        ? `${replacementSelectorPrefix} .`
+                        : `${replacementSelectorPrefix}.`,
+                    )
+                  ) {
+                    const atomicClassesArgs = [
+                      memberExpressionIndex,
+                      1,
+                      ...atomicClasses.map((atomicClass) =>
+                        j.memberExpression(
+                          memberExpression.value.object,
+                          j.identifier(atomicClass),
+                        ),
+                      ),
+                    ];
+                    parent.expressions.splice(...atomicClassesArgs);
+
+                    if (replacementSelector.includes(' > ')) {
+                      const quasisArgs = [
+                        memberExpressionIndex,
+                        1,
+                        j.templateElement(
+                          {
+                            raw: precedingTemplateElement.value.raw.replace(' ', ''),
+                            cooked: precedingTemplateElement.value.cooked.replace(' ', ''),
+                          },
+                          false,
+                        ),
+                        j.templateElement({ raw: ' > .', cooked: ' > .' }, false),
+                      ];
+
+                      if (atomicClasses.length === 3) {
+                        quasisArgs.splice(
+                          3,
+                          0,
+                          j.templateElement({ raw: '.', cooked: '.' }, false),
+                        );
+                      }
+
+                      parent.quasis.splice(...quasisArgs);
+                    } else {
+                      parent.quasis.splice(
+                        memberExpressionIndex,
+                        1,
+                        j.templateElement(
+                          {
+                            raw: precedingTemplateElement.value.raw,
+                            cooked: precedingTemplateElement.value.cooked,
+                          },
+                          false,
+                        ),
+
+                        j.templateElement({ raw: '.', cooked: '.' }, false),
+                      );
+                    }
+                  }
+                }
+              });
+          }
+        });
+      });
+
+    const selector = `${replacementSelectorPrefix}${deprecatedClass}`;
```

Comment:
Observed an issue with regex approach here (the approach which already merged PR's are using). 

when running codemod for ` .MuiButtonGroup-groupedTextHorizontal` regex matched with ` .MuiButtonGroup-groupedText` and applied replacementSelector of ` .MuiButtonGroup-groupedText` which is `.MuiButtonGroup-text > .MuiButtonGroup-grouped` instead of applying ` .MuiButtonGroup-groupedTextHorizontal`.

So went with absolute equality approach instead of regex equality

Reply:
The problem with absolute equality is that it won't match when the literal has more classes, for example `.Mui-disabled .MuiButtonGroup-groupedTextHorizontal`. We could use regex's `$` matcher to match the end of the file, or a dot `.` which would denote another class.

```js
new Regex(`${replacementSelectorPrefix}${deprecatedClass}(.|$)`)
```

Would that work?

Reply:
>The problem with absolute equality is that it won't match when the literal has more classes, for example .Mui-disabled .MuiButtonGroup-groupedTextHorizontal. We could use regex's $ matcher to match the end of the file, or a dot . which would denote another class.

i see, got it.

>Would that work?

yes, this worked. commit: https://github.com/mui/material-ui/pull/41259/commits/8d0f26cb41149d00f3a38ca2018bbb1b9832c355

Reply:
I'm sorry, my mistake. We must escape the dot as it's a regex symbol for any character. This regex should work:

```js
new Regex(`${replacementSelectorPrefix}${deprecatedClass}(\.| |$)`)
```

I added the space as well `" "` to accommodate for other selectors.

Let me know if that works and sorry for the late reply.

Reply:
Tried `new Regex(`${replacementSelectorPrefix}${deprecatedClass}(\.| |$)`)` and `new Regex(`${replacementSelectorPrefix}${deprecatedClass}(.| |$)`)` both are transforimg `('& .MuiButtonGroup-groupedTextHorizontal');` to `("&.MuiButtonGroup-text > .MuiButtonGroup-groupedorizontal");`

Reply:
That's weird, do you think it's matching with `MuiButtonGroup-groupedText` instead? It shouldn't:

<img width="1357" alt="Screenshot 2024-03-07 at 15 32 02" src="https://github.com/mui/material-ui/assets/16889233/97e4a68a-6090-4d59-b3db-a5f5007a5302">

Did this one work?:

```js
new RegExp(`${replacementSelectorPrefix}${deprecatedClass}$`)
```

Reply:
>Did this one work? new RegExp(`${replacementSelectorPrefix}${deprecatedClass}$`)

Yes, this worked. commit reference: https://github.com/mui/material-ui/pull/41259/commits/67e475f6b68ec100cafb1aebfff41e883a0c2b7f